### DISCUSSION TITLE
How to reduce the compilation time of ocelot.json with multiple configuration files?

### DISCUSSION THREAD

**Discussion by Erol5678** [ThreeMammals/Ocelot#1889] [Q&A] [answered]
## Expected Behavior / New Feature
The ocelot should not take much time at startup when we add multiple configuration files in gateway.

## Actual Behavior / Motivation for New Feature

Due to the multiple and large file size of configuration files, ocelot taking too much time at startup 
## Steps to Reproduce the Problem

  1.add multiple configuration files in ocelot with diffrent name 
  2  A single  file size could be of 84 kb 
  3 Start the ocelot gateway and observe how much time it will take to compile those files

## Specifications

  - Version: 
  - Platform:
  - Subsystem:

**Comment by raman-m** [comment, marked as answer]
How large are your files? How many files?
Why do you have these multiple config files?

If your config is really large, then seems service definitions should be managed by a service discovery tools like Consul. After that you'll have single config file with Consul service keys. One single file! No need to merge!

**Comment by raman-m** [comment]
> 1.add multiple configuration files in ocelot with diffrent name

Impossible to reproduce the issue!
